avr/cores/megacommand/MD/MD.cpp

```diff
@@ -162,10 +162,10 @@ void MDClass::triggerTrack(uint8_t track, uint8_t velocity) {
 }
 
 void MDClass::setTrackParam(uint8_t track, uint8_t param, uint8_t value) {
-  if (global.baseChannel > 15)
-    return;
-  if ((track > 15) || (param > 33))
+/*  if (global.baseChannel > 15)
     return;
+    if ((track > 15) || (param > 33))
+    return;*/
 
   uint8_t channel = track >> 2;
```

Comment:
track / 4? now that we know the trick ;)

Reply:
Did I miss something important. I remember you said that 
n / 4 didn't optimize to n >> 2

If you can see optimisation improvements for setTrackParam please assist :smiley:

Reply:
It's not "n/4 didn't optimize to n>>2", rather the opposite:
n/4 will perform better than n>>2 (!)

Reply:
>improvements for setTrackParam

sure, will go through that

Reply:
@yatli I worked out what's happening

Both n /4 and n >> 2 will be optimized to the same instruction set when the
-Os compiler flag is enabled

https://godbolt.org/z/sRIPhS

Reply:
we are already compiling with -Os

Reply:
This one is good read. https://ww1.microchip.com/downloads/en/AppNotes/doc8453.pdf

Reply:
check this out: https://godbolt.org/z/luMJFr

Reply:
When the shifted value is used as addressing index (which must be 16bit), shift generates two more instruction than the division version.

(I've no idea why this is happening... But let's just sacrifice the young goat, for fundamentally technical reasons. :D)

Reply:
😲

Reply:
I just took another look:

```
1:      asr r31
        ror r30
        dec r18
        brne 1b
```

so it's not just two more instructions. it's a loop